src/FsAutoComplete/CodeFixes/ConvertTripleSlashCommentToXmlTaggedDoc.fs

```diff
@@ -0,0 +1,217 @@
+
+    not xmlDoc.IsEmpty && not summaryPresent
+  else
+    false
+
+let private isLowerAstElemWithPreXmlDoc input pos =
+  SyntaxTraversal.Traverse(
+    pos,
+    input,
+    { new SyntaxVisitorBase<_>() with
+        member _.VisitBinding(_, defaultTraverse, synBinding) =
+          match synBinding with
+          | SynBinding(xmlDoc = xmlDoc) when containsPosAndNotEmptyAndNotElaborated pos xmlDoc -> Some xmlDoc
+          | _ -> defaultTraverse synBinding
+
+        member _.VisitComponentInfo(_, synComponentInfo) =
+          match synComponentInfo with
+          | SynComponentInfo(xmlDoc = xmlDoc) when containsPosAndNotEmptyAndNotElaborated pos xmlDoc -> Some xmlDoc
+          | _ -> None
+
+        member _.VisitInheritSynMemberDefn(_, _, _, _, members, _) =
+          let isInLine c =
+            match c with
+            // ToDo not working
+            | SynMemberDefn.AutoProperty(xmlDoc = xmlDoc) when containsPosAndNotEmptyAndNotElaborated pos xmlDoc ->
+              Some xmlDoc
+            | SynMemberDefn.ImplicitCtor(xmlDoc = xmlDoc) when containsPosAndNotEmptyAndNotElaborated pos xmlDoc ->
+              Some xmlDoc
+            | _ -> None
+
+          members |> List.tryPick isInLine
+
+        member _.VisitRecordDefn(_, fields, _) =
+          let isInLine c =
+            match c with
+            | SynField(xmlDoc = xmlDoc) when containsPosAndNotEmptyAndNotElaborated pos xmlDoc -> Some xmlDoc
+            | _ -> None
+
+          fields |> List.tryPick isInLine
+
+        member _.VisitUnionDefn(_, cases, _) =
+          let isInLine c =
+            match c with
+            | SynUnionCase(xmlDoc = xmlDoc) when containsPosAndNotEmptyAndNotElaborated pos xmlDoc -> Some xmlDoc
+            | _ -> None
+
+          cases |> List.tryPick isInLine
+
+        member _.VisitEnumDefn(_, cases, _) =
+          let isInLine b =
+            match b with
+            | SynEnumCase(xmlDoc = xmlDoc) when containsPosAndNotEmptyAndNotElaborated pos xmlDoc -> Some xmlDoc
+            | _ -> None
+
+          cases |> List.tryPick isInLine
+
+        member _.VisitLetOrUse(_, _, defaultTraverse, bindings, _) =
+          let isInLine b =
+            match b with
+            | SynBinding(xmlDoc = xmlDoc) when containsPosAndNotEmptyAndNotElaborated pos xmlDoc -> Some xmlDoc
+            | _ -> defaultTraverse b
+
+          bindings |> List.tryPick isInLine
+
+        member _.VisitExpr(_, _, defaultTraverse, expr) = defaultTraverse expr
+
+        member _.VisitImplicitInherit(_, defaultTraverse, _, synArgs, _) = defaultTraverse synArgs
+
+        member _.VisitType(_, defaultTraverse, synType) = defaultTraverse synType
+
+        member _.VisitMatchClause(_, defaultTraverse, matchClause) = defaultTraverse matchClause
+
+        member _.VisitModuleDecl(_, defaultTraverse, synModuleDecl) = defaultTraverse synModuleDecl
+
+        member _.VisitPat(_, defaultTraverse, synPat) = defaultTraverse synPat }
```

Comment:
may be able to remove these since they're just calling the defaults?

Reply:
The `VisitExpr` member is needed to let nested let bindings work, but yes, the rest can go. Thanks